solvers/DPStokes/mobility.h

```diff
@@ -44,30 +44,110 @@ class DPStokes: public libmobility::Mobility{
     dpstokes = std::make_shared<uammd_dpstokes::DPStokesGlue>();
   }
 
+  double configure_grid_and_kernels_xy(Parameters ipar){
+    double hydroRadius = ipar.hydrodynamicRadius[0];
+
+    // STEP 1: set w & beta based on kernel type (Table 1/2)
+    double w = 6.0; // note: should be an int but I'm scared of the consequences
+    double beta = 1.714;
+
+    // STEP 2: use c(beta*w) polyfit to get c
+      // note: polyfit is to c(beta*w), not just beta
+    double c_beta = polyEval(cbetam, beta*w);
+    double h = hydroRadius / (w*c_beta); // compute h using h = Rh/(w*c) and nx = Lx/h
+    double nx_unsafe = dppar.Lx/h; // not necessarily fft friendly
+
+    // STEP 3: find candidates for fft friendly nx
+    std::vector<int> nx_safe = fft_friendly_sizes(floor(nx_unsafe), 100, 10);
+    int m = nx_safe.size();
+
+    std::vector<double> errors(m);
+    std::vector<double> h_candidates(m);
+    std::vector<double> beta_candidates(m);
+
+    int n = 1000;
+    double start = w; // interpolation range
+    double end = 3*w;
+    std::vector<double> err_axis(n); // creates a linspace vector
+    h = (end-start)/(n-1);
+    std::cout << h << std::endl;
+    for(int i = 0; i < n; i++){
+      err_axis[i] = start + i*h;
+    }
+
+    // STEP 4: for each candidate, compute corresponding h and use c^{-1} polyfits to get beta
+    // then, linearly interpolate the error spline using beta*w to get error
+    for(int i = 0; i < m; i++){
+      h_candidates[i] = dppar.Lx/(1.0*nx_safe[i]); // double cast
+      beta_candidates[i] = polyEval(cbetam_inv, hydroRadius/(w*h_candidates[i])) / w;
+
+      double beta_w = beta_candidates[i]*w;
+      if (beta_w < w || beta_w > 3*w){ // out of range of interpolation
+        errors[i] = -1;
+        continue;
+      }
+      errors[i] = linearInterp(errmw6,err_axis,beta_w);
+    }
+
+    // pick parameters with minimum error
+    int best_index = 0;
+    for(int i = 1; i < m; i++){
+      if(errors[i] != -1 && errors[i] < errors[best_index]){
+        best_index = i;
+      }
+    }
+
+    // set smallest h, beta
+    this->dppar.nx = nx_safe[best_index]; 
+    this->dppar.ny = nx_safe[best_index]; // assumes Lx=Ly
+    this->dppar.w = w;
+    this->dppar.beta = beta_candidates[best_index]*w;
+    h = h_candidates[best_index];
+
+    return h;
+  }
```

Comment:
I believe this should be a free-standing function that returns nx,NY, w, beta, hxy.